### ISSUE TITLE
Why compute gradient cost in this example?

### ISSUE THREAD

**Issue by ArnoVel** [getkeops/keops#44]
Hi,
I am referencing [this specific example](https://github.com/getkeops/keops/blob/0f49e01105ed6c586b775429a3a1fe7b239c84f6/pykeops/tutorials/gaussian_mixture/plot_gaussian_mixture.py#L27-L38) in which the goal is to fit a GMM with flexible number of mixtures to some 2d points denoted ```x```.

I do not understand the point of ```x.requires_grad = True``` here. In other examples, (shape matching) you do update the data values (x) so that the original sample matches some other target sample.
However here ```x``` is (if I'm correct) the target sample.

Removing this line with fixed seed yields the same results.

**Comment by jeanfeydy** [comment]
Hi @ArnoVel ,

Indeed you're right! Some copy-paste was involved in the creation of this 2D spiral, and I just forgot to remove the `requires_grad = True` line. Anyway: we will deprecate the `kernel_product` syntax soon (`LazyTensors` are just much simpler and faster) and rewrite the tutorial soon. I'll make sure to remove this small typo :-)

Best regards,
Jean
